spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:

```

Comment:
Do specs use the term `struct`?  Do you have examples of other specs that use this term?

Reply:
Ah it looks like they do use struct, https://infra.spec.whatwg.org/#structs
But should we be calling this a struct? https://html.spec.whatwg.org/#browsing-context doesn't call their thing a struct

Reply:
The  close-watcher spec uses struct:  https://github.com/WICG/close-watcher/blob/main/spec.bs

Reply:
However, since the idl for the same structures use dictionary, I changed the structs to dictionary.

Reply:
I think this should be a struct as it's part of the spec (so should use spec terms, not web idl terms as it's not web accessible) and I think it better matches the defintion of a struct, "finite set of items, each of which has a unique and immutable name"

Reply:
I'm not sure, but I don't think we want both struct and dictionary.  Jeffrey says, regarding this question:  "Ah, and you probably don't want to define parallel structs and dictionaries. The dictionary is sufficient for taking parameters to functions. You only need structs when you want to do internal processing, or store data for a while.
And generally the stored form wouldn't exactly match the dictionaries that are arguments to functions."

Reply:
Both of the data structures in this PR are only for internal processing, so they should be structs.

Reply:
I guess I don't feel strongly, but it seems like the members of this data structure are all fixed items with fixed names and types, which seems better suited to a struct rather than an open ended dictionary.

Reply:
Indeed, a struct sounds more appropriate than a dictionary.  But there is not an idl way to specify a struct.  The joinAdInterestGroup function references the idl for the AuctionAdInterestGroup definition, so this data structure is not only for internal processing.  

But I am still not clear why we need infra separate (different, and not one-to-one) from web idl.